packages/cli/src/start/server.ts

```diff
@@ -0,0 +1,199 @@
+import qrcode from 'qrcode-terminal'
+import { WebSocketServer } from 'ws'
+import z from 'zod'
+
+// TODO
+import { connectWebSocket, createWSServerContext } from '../../../server/src/ws-connection'
+import { IGNORED_PATH, WATCH_PATH } from '../config/constants'
+
+interface IndexedModels {
+  [key: string]: any
+}
+
+export const devArgsSchema = z.object({
+  ws: z.boolean().optional().default(false),
+  prod: z.boolean().optional().default(false),
+  port: z.number(),
+  host: z.string(),
+  cwd: z.string().optional(),
+})
+
+export type DevArgs = z.infer<typeof devArgsSchema>
+
+class ModelManager {
+  models: AnyModels & IndexedModels = {}
+  cwd: string = process.cwd()
+
+  constructor(cwd: string) {
+    this.cwd = cwd
+  }
+
+  parseModelPath(modelPath: string) {
+    const key = modelPath.split('/').slice(1, -1).join(':')
+    const path = modelPath.split('.').slice(0, -1).join('.')
+    return { key, path }
+  }
+
+  async updateModel(modelPath: string) {
+    const { key, path: basePath } = this.parseModelPath(modelPath)
+    const cwd = this.cwd!
+    this.models[key] = await import(path.join(cwd, basePath))
+  }
+
+  removeModel(modelPath: string) {
+    const { key } = this.parseModelPath(modelPath)
+    this.models[key]
+  }
+
+  getModels() {
+    return this.models
+  }
+}
+
+export class DevServer {
+  options: DevArgs
+  modelManager: ModelManager
+
+  constructor(options: DevArgs) {
+    this.options = options
+    this.modelManager = new ModelManager(options.cwd!)
+
+    this.start()
+  }
+
+  async start() {
+    const port = this.options.port
+    const host = this.getHost()
+
+    const fsWatcher = await this.watchFiles()
+
+    if (this.options.prod) {
+      // Close the watcher if we are in prod mode after adding all models
+      fsWatcher.close()
+    }
+
+    this.startServer()
+
+    console.log('Server started on', `${host}:${port}`)
+    console.log('Scan the QR from rise playground')
+
+    this.createDevQR(`rise-playground://${host}:${port}`)
+  }
+
+  getIpAddress(): string {
+    return internalIp.v4.sync() || '127.0.0.1'
+  }
+
+  getHost() {
+    switch (this.options.host) {
+      case 'localhost':
+        return 'localhost'
+      case 'lan':
+        return this.getIpAddress()
+    }
+  }
+
+  createDevQR(url: string) {
+    qrcode.generate(url, { small: true }, function (qrcode) {
+      console.log(qrcode)
+    })
+  }
+
+  httpServer(port: number) {
+    const modelManager = this.modelManager
+    const server = Fastify()
+
+    server.get('*', async function handler(request, reply) {
+      const path = request.url.split('/').filter(Boolean)
+      const models = modelManager.getModels()
+      const resp = await handleGetRequest(models, { path })
+      if (!resp) return reply.code(404).send({ error: 'not found' })
+      return resp
+    })
+
+    server.listen({ port }).then(() => {})
+    return {
+      close() {
+        server.close()
+      },
+    }
+  }
+
+  wsServer(port: number) {
```

Comment:
question: Why don't we just do:
```ts
return createWSServer({ models: this.modelManager.getModels(), port })
```
Was there any particular reason? Because if not - then, we don't need to import those "lower-level" functions from `ws-server`, but can use top-level export instead.

Reply:
Same goes for `httpServer` method. It looks very similar to `createHTTPServer` and I wonder if we could use that one directly instead.

Reply:
we also have `createServer()` which supports both HTTP and WS. Not sure why we wouldn't just always use that one

Reply:
I tried with hono initially, it's lightweight.